### DISCUSSION TITLE
Winpilot 2024.6.19

### DISCUSSION THREAD

**Discussion by Belim** [builtbybel/Bloatynosy#224] [Announcements]
## After I gave Winpilot ("WebApp") a modern look with a classic Win32 desktop app two weeks ago - the project ran in parallel and could optionally be downloaded as Clipilot - the feedback was overwhelming, and a [survey to decide whether to go with the WebApp or Win32](https://www.builtbybel.com/blog/future-direction-for-winpilot-web-app-or-win32-desktop-app) is now almost unnecessary.

I received over 130 messages via email alone - that really wasn't necessary, folks, but yes, I agree with you - I should indeed strip the web core from Winpilot.

Given that Microsoft has recently taken the "wrong" direction again by turning its nice (not perfect but good) UWP-era Mail and Calendar app into a PWA, I wasn't particularly thrilled, to put it mildly. Quite the opposite, I will banish this app in the business environment. Apart from the fact that the app is a resource monster - PWAs rely heavily on JavaScript, which is more resource-intensive than native code, especially for complex applications like web clients - this app stores login data for IMAP accounts on Microsoft's servers, and that's simply too much for me!

Well, I don't want to digress. Winpilot is getting rid of the Edge WebView2 foundation and will return to being a classic application. **Don't be surprised if "Clipilot" appears in the title bar at startup; this is the start module of the Clippy assistant. All other classic modules are named Winpilot, just like the app itself.** 

Here are some advantages of the new "old" classic Win32 desktop app
- **Transitioned the app to a "native" implementation, removing the need for WebView2**
- The app operates with minimal resource consumption and loads up to 70% faster
- **The app utilizes optimized yet classic Bloatynosy modules. The first two (CoTweaker Setup and Decrapify) have already been completely rewritten**
- Improved logging and system monitoring through Clippy; all settings are logged in the Clippy Bubble
- The app no longer requires admin rights and must be manually granted, as Clippy prefers it that way
- The Clippy character/appearance can be customized
- **Clippy can also be disabled**
- No Copilot support for now, but we may consider integrating Copilot with Clippy as our mascot in the future

<hr /><em>This discussion was created from the release <a href='https://github.com/builtbybel/Winpilot/releases/tag/2024.6.19'>Winpilot 2024.6.19</a>.</em>

**Comment by azonin** [comment]
Installed and running the 2024.6.19 version. Is the "Setup" tab supposed to have some "Apply" or "Run" button anywhere? There's a bunch of checkboxes, but they don't do anything, even after reboot - the Win 11 Co-Pilot is still hanging in the systray, no matter if I enable or disable it in the Setup tab.

I'm running 3840x2160 at 100% Scale. Tried to resize the window - but I still don't see how to apply the selected settings.

**Reply by Belim** [reply]
That's correct, there is no apply button. Actions are set directly with a check/uncheck. Some settings require administrative rights.

**Reply by azonin** [reply]
I'm running it as admin, but some of these changes still do not apply, even after full reboot. For example, the "AI/Copilot > Copilot in Taskbar" erroneously sets the "HKEY_CURRENT_USER\Software\Policies\Microsoft\Windows\TurnOffWindowsCopilot" to 1, which doesn't do anything. It should be setting it to 1 in HKEY_CURRENT_USER\Software\Policies\Microsoft\Windows\WindowsCopilot\TurnOffWindowsCopilot and in HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\WindowsCopilot\TurnOffWindowsCopilot

What does "Potential issues" checkbox do? There's no tooltip above it.

The tooltips above each Setup option are useful, but you have to hover each one to see if it says "This feature will disable" or "This feature will enable". I think it would be more intuitive to simply start each option with the words "Disable ..." and "Enable ...", so you don't have to hover.

I'd also add a description above this section saying whether these options are applied to the registry instantly, or after you get out of this screen, or after you close Winpilot. Otherwise it's unclear, as the "Debloat" tab has an "Apply" button, which is a different UX pattern.

**Reply by Belim** [reply]
I have reviewed numerous settings again, especially HKEY_CURRENT_USER\Software\Policies\Microsoft\Windows\WindowsCopilot\TurnOffWindowsCopilot, and the switch works perfectly. You might want to try the Winpilot Legacy https://github.com/builtbybel/Winpilot/releases/download/2024.6.19/Winpilot_Legacy.WebApp.zip version, the settings are the same, and you can check if the switch gets toggled. The **Potential Issues** node is simply the root node; it might need a better name, as you suggested. We could introduce a button, but I wanted to keep it simpler and just allow checking or unchecking.

**Reply by azonin** [reply]
> I have reviewed numerous settings again, especially HKEY_CURRENT_USER\Software\Policies\Microsoft\Windows\WindowsCopilot\TurnOffWindowsCopilot, and the switch works perfectly. You might want to try the Winpilot Legacy https://github.com/builtbybel/Winpilot/releases/download/2024.6.19/Winpilot_Legacy.WebApp.zip version, the settings are the same, and you can check if the switch gets toggled. The **Potential Issues** node is simply the root node; it might need a better name, as you suggested. We could introduce a button, but I wanted to keep it simpler and just allow checking or unchecking.

I believe you had the "WindowsCopilot" missing from the path.

**Reply by Belim** [reply]
Ahh, damn, it is indeed HKCU\Software\Policies\Microsoft\Windows\WindowsCopilot. I'm surprised no one noticed it until today, thanks a lot

**Reply by azonin** [reply]
I'd love to collaborate on the UX/IxDA with you - let me know if you're interested.

**Reply by Belim** [reply]
Sure, I'm always interested in feedback and exchange, especially when it comes to UI. I'm watching with interest. I need to clean up and reorganize the repo soon and push the current code.
